src/Authentication/Authenticator.php

```diff
@@ -134,6 +134,17 @@ function getProfile()
     }
 
     /**
+     * will return a valid userProfile object for a pre-loaded profile
+     * (non-PHPdoc)
+     *
+     * @see \PhalconRest\Authentication\AuthenticatorInterface::getProfile()
+     */
+    function setProfile(\PhalconRest\Authentication\UserProfile $profile)
+    {
+        return $this->profile = $profile;
+    }
```

Comment:
well, turns out this is not needed and the "hole" I envisioned is already present on the API. You can use loadProfile() from the Authenticator component to achieve the same result without adding a new method that overrides the already created object :)

Remove this method and I'll merge the PR.

Will explain @ Bitbucket.